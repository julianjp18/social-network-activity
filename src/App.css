.post-list-container {
  padding: 20px;
}

.post-list-container .button-container {
  text-align: center;
}

.comments-container {
  margin: auto;
}

.ant-menu-root {
  position: fixed;
  width: 23%;
}