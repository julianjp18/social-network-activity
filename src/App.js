import React, { useEffect } from "react";
import {
  ADD_COMMENT,
  ALL_COMMENTS,
  GET_COMMENTS,
  GET_POSTS,
} from "./redux/constants";
import axios from "axios";
import { BrowserRouter as Router, Switch, Route, Redirect } from "react-router-dom";
import { Col, Row } from "antd";
import { connect } from "react-redux";
import Comments from "./components/Comments/Comments";
import { func } from "prop-types";
import Menu from "./components/Menu/Menu";
import { object } from "prop-types";
import Posts from "./components/Posts/Posts";
import "antd/dist/antd.css";
import "./App.css";

const App = ({
  addCommentAction,
  allComments,
  allCommentsAction,
  postComments,
  posts,
  postsAction,
  postCommentsAction,
}) => {
  useEffect(() => {
    axios.get(`https://jsonplaceholder.typicode.com/comments`).then((res) => {
      if (res.status === 200) {
        allCommentsAction(res.data);
      }
    });
  }, [allCommentsAction]);

  useEffect(() => {
    axios.get(`https://jsonplaceholder.typicode.com/posts`)
      .then(res => {
        if (res.status === 200)
          postsAction(res.data);
      })
  }, [postsAction]);

  return (
    <Router>
      <Row>
        <Col xs={6}>
          <Menu />
        </Col>
        <Col xs={18}>
          <Row className="post-list-container">
            <Col xs={24}>
              <h1>Social network activity - recently posts</h1>
            </Col>
            <Col xs={24}>
              <Switch>
                <Route path="/comments/:postId">
                  <Comments
                    addCommentAction={addCommentAction}
                    allComments={allComments}
                    comments={postComments}
                    postCommentsAction={postCommentsAction}
                  />
                </Route>
                <Route path="/posts">
                  <Posts allComments={allComments} posts={posts} postsAction={postsAction} />
                </Route>
                <Redirect to="/posts" />
                <Route exact path="/">
                  <Redirect to="/posts" />
                </Route>
              </Switch>
            </Col>
          </Row>
        </Col>
      </Row>
    </Router>
  );
};

App.propTypes = {
  addComment: func.isRequired,
  comments: object.isRequired,
  posts: object.isRequired,
};

App.defaultProps = {};

const mapStateToProps = (state) => ({
  posts: state.posts,
  postComments: state.post_comments,
  allComments: state.all_comments,
});

const mapDispatchToProps = (dispatch) => {
  return {
    addCommentAction: (payload) => dispatch({ type: ADD_COMMENT, payload }),
    allCommentsAction: (payload) => dispatch({ type: ALL_COMMENTS, payload }),
    postCommentsAction: (payload) => dispatch({ type: GET_COMMENTS, payload }),
    postsAction: (payload) => dispatch({ type: GET_POSTS, payload }),
  };
};
export default connect(mapStateToProps, mapDispatchToProps)(App);
