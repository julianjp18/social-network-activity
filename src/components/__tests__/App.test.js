import React from 'react';
import ReactDOM from 'react-dom';
import Adapter from 'enzyme-adapter-react-16';
import App from '../../App';
import Enzyme, { shallow, render, mount } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() })

describe('App tests', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Provider store={store}><App/></Provider>)
  
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});