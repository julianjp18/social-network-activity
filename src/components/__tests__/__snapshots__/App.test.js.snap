// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(App) />
</ContextProvider>
`;
